src/main/groovy/net/minecraftforge/gradleutils/GradleUtilsExtension.groovy

```diff
@@ -5,138 +5,217 @@
+    }()
 
     @Inject
-    GradleUtilsExtension(Project project) {
+    GradleUtilsExtension(Project project, Problems problems) {
         this.project = project
+        this.problems = problems
+        this.pom = new PomUtils(problems, project.extensions.getByType(GitVersionExtension))
+
+        this.setupTasks()
+
+        this.gitRoot = project.objects.directoryProperty().convention project.layout.projectDirectory
+    }
+
+    private void setupTasks() {
+        // Actions Workflow Generator
+        GenerateActionsWorkflow.register(this.project)
 
-        this.gitRoot = project.objects.directoryProperty().convention(project.layout.projectDirectory)
-        this.gitInfo = project.objects.mapProperty(String, String)
-                .convention(gitRoot.map((Directory dir) -> GradleUtils.gitInfo(dir.asFile)))
+        // Legacy CI Tasks
+        GradleUtils.setupCITasks(this.project)
+    }
+
+    void requestTask(TaskProvider<? extends Task> task) {
+        this.project.gradle.startParameter.taskRequests = new ArrayList<TaskExecutionRequest>(this.project.gradle.startParameter.taskRequests).tap {
+            it.add(new SimpleTaskExecutionRequest(task))
+        }
     }
 
     /**
-     * Returns a version in the form {@code $tag.$offset}, e.g. 1.0.5
+     * This method has been deprecated in favor of usage of GitVersion.
+     * <pre><code>
+     *     // Before:
+     *     version = gradleutils.tagOffsetVersion
      *
-     * @return a version in the form {@code $tag.$offset}, e.g. 1.0.5
+     *     // After:
+     *     version = gradleutils.version.tagOffset
+     * </code></pre>
+     *
+     * @deprecated Use {@link net.minecraftforge.gitver.api.GitVersion#getTagOffset() GitVersion.tagOffset} instead.
      */
+    @Deprecated(forRemoval = true, since = '2.4')
     String getTagOffsetVersion() {
-        return GradleUtils.getTagOffsetVersion(getGitInfo())
+        this.project.extensions.getByType(GitVersionExtension).version.tagOffset
     }
 
     /**
-     * Returns a version in the form {@code $tag.$offset}, e.g. 1.0.5.
-     * The provided filter is used to filter the retrieved tag.
+     * This method has been deprecated in favor of usage of GitVersion.
+     * <pre><code>
+     *     // Before (using filter):
+     *     version = gradleutils.getFilteredTagOffsetVersion('1.21.*')
+     *
+     *     // After (using filter):
+     *     gradleutils.version.addMatchFilters('1.21.*')
+     *     version = gradleutils.version.tagOffset
      *
-     * @param prefix If true, will treat the filter as a prefix.
-     * Defaults to false, which means to treat the filter as a glob pattern.
-     * @param filter A non-null string filter used when retrieving the tag
-     * @return a version in the form {@code $tag.$offset}, e.g. 1.0.5
+     *     // After (using tag prefix):
+     *     gradleutils.version.tagPrefix = '1.21.x' // = '1.21.x-*'
+     *     version = gradleutils.version.tagOffset
```

Comment:
Could we not keep the old syntax by just adding a bouncer method that does these two lines?

Reply:
Ah, actually I forgot to update these docs. As of right now, tag prefixes can't be updated at all since configs are immutable and read from the `.gitversion` file in GitVersion 0.2. I can change that if need be, but I'd prefer to keep it that way.

Reply:
I'm still torn on this. Should I just add some temporary, deprecated setter methods in Git Version to allow for this? At the end of the day when Git Version 1.0 is finished, I would prefer that the config (includes tag prefix and filters) would be data-only, and be unmodifiable in code.

Reply:
Is this used much in the existing org? Probably not worth it.

Reply:
I decided to add the setter methods for now in Git Version 0.2.6, just to reserve back-compat for this. Off the top of my head: the repos that use this are Forge and ForgeGradle (I think???). But, I don't want to take any chances. I can just remove them later.